src/rgw/rgw_rados.cc

```diff
@@ -4449,20 +4449,20 @@ int RGWRados::init_complete()
     obj_expirer->start_processor();
   }
 
-  if (run_sync_thread) {
-    // initialize the log period history. we want to do this any time we're not
-    // running under radosgw-admin, so we check run_sync_thread here before
-    // disabling it based on the zone/zonegroup setup
-    meta_mgr->init_oldest_log_period();
-  }
-
   /* no point of running sync thread if we don't have a master zone configured
     or there is no rest_master_conn */
   if (get_zonegroup().master_zone.empty() || !rest_master_conn
       || current_period.get_id().empty()) {
     run_sync_thread = false;
   }
 
+  if (run_sync_thread) {
+    // initialize the log period history. we want to do this any time we're not
+    // running under radosgw-admin, so we check run_sync_thread here before
+    // disabling it based on the zone/zonegroup setup
+    meta_mgr->init_oldest_log_period();
+  }
```

Comment:
moving this down is fine, but the comment still talks about checking `run_sync_thread here before disabling it based on the zone/zonegroup setup`, which happens in the block above - could you please update or remove the comment?

Reply:
sure